samcli/cli/cli_config_file.py

```diff
@@ -145,44 +165,88 @@ def configuration_option(*param_decls, **attrs):
             def hello(name):
                 print("Hello " + name)
 
-    This will create an option of type `STRING` expecting the config_env in the
-    configuration file, by default this config_env is `default`. When specified,
-    the requisite portion of the configuration file is considered as the
-    source of truth.
-
-    The default name of the option is `--config-env`.
-
-    This decorator accepts the same arguments as `click.option`.
-    In addition, the following keyword arguments are available:
-    :param cmd_name: The command name. Default: `ctx.info_name`
-    :param config_env_name: The config_env name. This is used to determine which part of the configuration
-        needs to be read.
-    :param provider:         A callable that parses the configuration file and returns a dictionary
+    By default, this will create a hidden click option whose callback function loads configuration parameters from
+    default configuration environment [default] in default configuration file [samconfig.toml] in the template file
+    directory.
+    :param preconfig_decorator_list: A list of click option decorator which need to place before this function. For
+        exmple, if we want to add option "--config-file" and "--config-env" to allow customized configuration file
+        and configuration environment, we will use configuration_option as below:
+        @configuration_option(
+            preconfig_decorator_list=[decorator_customize_config_file, decorator_customize_config_env],
+            provider=TomlProvider(section=CONFIG_SECTION),
+        )
+        By default, we enable these two options.
+    :param provider: A callable that parses the configuration file and returns a dictionary
         of the configuration parameters. Will be called as
         `provider(file_path, config_env, cmd_name)
     """
-    param_decls = param_decls or ("--config-env",)
-    option_name = param_decls[0]
-
-    def decorator(f):
-
-        attrs.setdefault("is_eager", True)
-        attrs.setdefault("help", "Read config-env from Configuration File.")
-        attrs.setdefault("expose_value", False)
-        # --config-env is hidden and can potentially be opened up in the future.
-        attrs.setdefault("hidden", True)
-        # explicitly ignore values passed to --config-env, can be opened up in the future.
-        config_env_name = DEFAULT_ENV
+
+    def decorator_configuration_setup(f):
+        configuration_setup_param_devls = ()
```

Comment:
what does `configuration_setup_param_devls` mean, the meaning is not apparent from the name.

Reply:
It is the parameters of this decorator, as it does not accept any parameters from command line. I put it empty. The name has been changed to "configuration_setup_params".